### DISCUSSION TITLE
Get the value of input names as text?

### DISCUSSION THREAD

**Discussion by aegis72** [Comfy-Org/ComfyUI#1511] [Q&A] [answered]
A picture means a thousand words they say... so I am trying to get the titles of the inputs in my node here, and use them as text to build a filename. This is part of a much larger node structure that makes many versions of an image (depth, normal, highlight, shadow, etc) and saves them all out to a nice organized folder and filename format automatically. 

Once a generation is worth it, I want to send it through the upscaler, and then have the option chosen here added to a text node that I can concat with "_upscaled" 

![image](https://github.com/comfyanonymous/ComfyUI/assets/118572301/fc72ffd1-d728-4954-b119-4d919233041f)

How might I access these nameson the left (not the image inputs, just the text of "processed, unprocessed, lineart, and depth?)

**Comment by ltdrdata** [comment, marked as answer]
The latest version of the `Impact Pack` has added this feature to `ImpactSwitch`.

![impactswitch](https://github.com/comfyanonymous/ComfyUI/assets/128333288/1083488d-ab40-454d-9e12-4a455c9d1b52)

**Comment by donengel72** [comment]


**Reply by aegis72** [reply]
Worked like a charm and eliminated a bunch of noodles. Thanks!
![image](https://github.com/comfyanonymous/ComfyUI/assets/118572301/220e29a7-562a-4346-854f-112d45e7f013)
